src/App.tsx

```diff
@@ -15,33 +15,72 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPHA,
+  LENGTH,
+}
+
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer];
+  const [isStarted, switchStart] = useState(false);
+  const [sortBy, setSortBy] = useState(SortType.NONE);
+  const [isReversed, switchReverse] = useState(false);
+  const reset = () => {
+    switchReverse(false);
+    setSortBy(SortType.NONE);
+  };
+
+  goods.sort((g1, g2) => {
+    switch (sortBy) {
+      case SortType.ALPHA:
+        return g1.localeCompare(g2);
+
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="App">
+      {isStarted
+        ? (
+          <>
+            <button type="button" onClick={() => setSortBy(SortType.ALPHA)}>
+              Sort alphabetically
+            </button>
```

Comment:
```suggestion
            <button
              type="button"
              onClick={() => setSortBy(SortType.ALPHA)}
            >
              Sort alphabetically
            </button>
```

Reply:
suggestion: when passing callbacks to `onClick` and so on, I'd rather create separate function-handlers